src/main/java/org/folio/service/orders/lines/update/OrderLinePatchOperationService.java

```diff
@@ -118,39 +136,94 @@ private Future<Void> sendPatchOrderLineRequest(OrderLineUpdateInstanceHolder ord
     return Future.succeededFuture();
   }
 
-  private PoLine updatePoLineWithInstanceRecordInfo(JsonObject lookupObj, PoLine poLine) {
-    Pair<String, String> instancePublication = Optional.ofNullable(lookupObj.getJsonArray(INSTANCE_PUBLICATION)).orElse(new JsonArray())
+  private Future<PoLine> updatePoLineWithInstanceRecordInfo(JsonObject lookupObj, PoLine poLine, RequestContext requestContext) {
+    Promise<PoLine> promise = Promise.promise();
+    Pair<String, String> instancePublication = getInstancePublication(lookupObj);
+    List<ProductId> productIds = getProductIds(lookupObj);
+
+    poLine.setTitleOrPackage(lookupObj.getString(INSTANCE_TITLE));
+    poLine.setPublisher(instancePublication.getLeft());
+    poLine.setPublicationDate(instancePublication.getRight());
+    poLine.setContributors(getContributors(lookupObj));
+
+    inventoryCache.getProductTypeUuid(requestContext)
+      .compose(isbnTypeId -> {
+        Semaphore semaphore = new Semaphore(SEMAPHORE_MAX_ACTIVE_THREADS, requestContext.getContext().owner());
```

Comment:
Probably this utility method can be used here, and it will be used for all futher usages:
https://github.com/folio-org/mod-invoice/pull/413/files#diff-c96a9c3d304e14f8c42a8d3eb5e849201c22e54098cd68d31316d2f0d38d657eR175